qualtran/bloqs/generalized_qsp_test.py

```diff
@@ -144,24 +143,35 @@ def verify_generalized_qsp(U: GateWithRegisters, P: Sequence[complex]):
 
     expected_top_left = evaluate_polynomial_of_matrix(P, input_unitary)
     actual_top_left = result_unitary[:N, :N]
-    assert_matrices_same_upto_global_phase(expected_top_left, actual_top_left)
+    assert_matrices_almost_equal(expected_top_left, actual_top_left)
 
     expected_bottom_left = evaluate_polynomial_of_matrix(gqsp_U.Q, input_unitary)
     actual_bottom_left = result_unitary[N:, :N]
-    assert_matrices_same_upto_global_phase(expected_bottom_left, actual_bottom_left)
+    assert_matrices_almost_equal(expected_bottom_left, actual_bottom_left)
 
 
 @pytest.mark.parametrize("bitsize", [1, 2, 3])
-@pytest.mark.parametrize("degree", [2, 3, 4])
+@pytest.mark.parametrize("degree", [2, 3, 4, 5, 50, 100, 150, 180])
 def test_generalized_real_qsp_on_random_unitaries(bitsize: int, degree: int):
     random_state = np.random.RandomState(42)
 
-    for _ in range(20):
+    for _ in range(10):
         U = RandomGate.create(bitsize, random_state=random_state)
         P = random_qsp_polynomial(degree, random_state=random_state, only_real_coeffs=True)
         verify_generalized_qsp(U, P)
 
 
+@pytest.mark.parametrize("bitsize", [1, 2, 3])
+@pytest.mark.parametrize("degree", [2, 3, 4, 5, 50, 100, 120])
+def test_generalized_qsp_on_random_unitaries(bitsize: int, degree: int):
```

Comment:
This test is for general complex unitaries whereas the one above is for real unitaries only, correct?
```suggestion
def test_generalized_qsp_on_random_complex_unitaries(bitsize: int, degree: int):
```

Reply:
No no, both use complex unitaries. The first test is for P with only real coeffs, and second is complex coeffs.

Reply:
Yeah sorry, I meant real vs complex polynomials. Maybe change the name of the test appropriately.

Reply:
updated